pipeline/metadata/schema.py

```diff
@@ -0,0 +1,299 @@
+"""Schema for various data types passed around in the pipeline and bigquery"""
+
+from __future__ import absolute_import
+from __future__ import annotations  # required to use class as a type inside the class
+
+import dataclasses
+from dataclasses import dataclass
+from typing import Optional, List, Dict, Any
+
+# pylint: disable=too-many-instance-attributes
+
+
+@dataclass
+class SatelliteAnswer():
+  """Class for keeping track of Satellite answer content"""
+  # Keys
+  ip: str
+  # Metadata
+  asnum: Optional[int] = None
+  asname: Optional[str] = None
+  http: Optional[str] = None
+  cert: Optional[str] = None
+  matches_control: Optional[str] = None
+
+  def add_tag_to_answer(self, tag: SatelliteAnswerMetadata) -> None:
+    """Add tag information to a Satellite answer"""
+    if tag.asnum is not None:
+      self.asnum = tag.asnum
+    if tag.asname is not None:
+      self.asname = tag.asname
+    if tag.http is not None:
+      self.http = tag.http
+    if tag.cert is not None:
+      self.cert = tag.cert
+    if tag.matches_control is not None:
+      self.matches_control = tag.matches_control
+
+
+@dataclass
+class SatelliteAnswerMetadata(SatelliteAnswer):
+  """Satellite Answer Metadata.
+
+  When this metadata is being passed around
+  it needs an additional date field to keep track of when it's valid.
+  In the final SatelliteAnswer object written to bigquery
+  we don't include that field since it's redundant.
+  """
+  date: str = ''
+
+
+@dataclass
+class IpMetadata():
+  """Class for keeping track of ip metadata"""
+  # Keys
+  ip: str
+  date: str
+  # Metadata
+  netblock: Optional[str] = None
+  asn: Optional[int] = None
+  as_name: Optional[str] = None
+  as_full_name: Optional[str] = None
+  as_class: Optional[str] = None
+  country: Optional[str] = None
+  organization: Optional[str] = None
+  # Satellite Metadata
+  name: Optional[str] = None
```

Comment:
Maybe rename to `hostname`

Reply:
Agreed, but this schema has to match the schema in bigquery, which should change in [PR119](https://github.com/censoredplanet/censoredplanet-analysis/pull/119), not here.

Reply:
This is a clear example of how the lack of a serialization layer prevents us from making changes